prime-router/src/test/kotlin/fhirengine/utils/FHIRBundleHelpersTests.kt

```diff
@@ -782,4 +783,90 @@ class FHIRBundleHelpersTests {
         assertThat(extension.url).isEqualTo(conditionCodeExtensionURL)
         assertThat((extension.value as? Coding)?.code).isEqualTo("6142004")
     }
+
+    @Test
+    fun `addMappedCondition supports processing values other than codeable concept`() {
+        val metadata = Metadata(UnitTestUtils.simpleSchema)
+
+        metadata.lookupTableStore += mapOf(
+            "observation-mapping" to LookupTable(
+                "observation-mapping",
+                listOf(
+                    listOf(
+                        ObservationMappingConstants.TEST_CODE_KEY,
+                        ObservationMappingConstants.CONDITION_CODE_KEY,
+                        ObservationMappingConstants.CONDITION_CODE_SYSTEM_KEY,
+                        ObservationMappingConstants.CONDITION_NAME_KEY
+                    ),
+                    listOf(
+                        "80382-5",
+                        "6142004",
+                        "SNOMEDCT",
+                        "Influenza (disorder)"
+                    ),
+                    listOf(
+                        "260373001",
+                        "Some Condition Code",
+                        "Condition Code System",
+                        "Condition Name"
+                    )
+                )
+            )
+        )
+
+        val entry = Observation()
+        val code = CodeableConcept()
+        code.addCoding(Coding("system", "80382-5", "display"))
+        entry.setCode(code)
+
+        entry.setValue(StringType("A string value"))
+
+        entry.addMappedCondition(metadata)
+
+        val extension = code.coding.first().extension.first()
+        assertThat(extension.url).isEqualTo(conditionCodeExtensionURL)
+        assertThat((extension.value as? Coding)?.code).isEqualTo("6142004")
```

Comment:
You can chain these assertions to have better output if it were to fail.

```kotlin
assertThat(extension.value)
  .isInstanceOf<Coding>()
  .transform { it.code }
  .isEqualTo("6142004")
```

If `extension.value` changes its type you would get a better error message than what would currently happen which is saying something like `null does not equal "6142004"`.

Reply:
Oh that's neat, will change that